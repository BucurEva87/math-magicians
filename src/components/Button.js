import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';

export default class Button extends PureComponent {
  render() {
    const {
      id, className, children, handleClick,
    } = this.props;

    return (
      <button
        type="button"
        id={id}
        className={className}
        onClick={handleClick}
      >
        {children}
      </button>
    );
  }
}

Button.defaultProps = {
  className: '',
};

Button.propTypes = {
  id: PropTypes.string.isRequired,
  className: PropTypes.string,
  children: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
};
