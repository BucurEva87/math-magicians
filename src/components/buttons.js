export default [
  { id: 'allClear', text: 'AC' },
  { id: 'switchSign', text: '+/-' },
  { id: 'percent', text: '%' },
  { id: 'divide', text: 'รท', className: 'operation' },
  { id: 'btn7', text: '7' },
  { id: 'btn8', text: '8' },
  { id: 'btn9', text: '9' },
  { id: 'times', text: 'x', className: 'operation' },
  { id: 'btn4', text: '4' },
  { id: 'btn5', text: '5' },
  { id: 'btn6', text: '6' },
  { id: 'minus', text: '-', className: 'operation' },
  { id: 'btn1', text: '1' },
  { id: 'btn2', text: '2' },
  { id: 'btn3', text: '3' },
  { id: 'plus', text: '+', className: 'operation' },
  { id: 'btn0', text: '0' },
  { id: 'dot', text: '.' },
  { id: 'equals', text: '=', className: 'operation' },
];
