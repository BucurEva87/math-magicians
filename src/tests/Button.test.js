import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Button from '../components/Button';
import buttons from '../components/buttons';

describe('Calculator interface buttons', () => {
  buttons.forEach((button) => {
    const { id, text } = button;

    test(`${id} button renders on the screen`, () => {
      const { getByTestId } = render(<Button id={id} handleClick={() => {}}>{text}</Button>);

      expect(getByTestId(id)).toBeInTheDocument();
    });
  });

  buttons.forEach((button) => {
    const { id, text } = button;

    test(`${id} button to have ${text} text`, () => {
      const { getByTestId } = render(<Button id={id} handleClick={() => {}}>{text}</Button>);

      expect(getByTestId(id)).toHaveTextContent(text);
    });
  });

  buttons.forEach((button) => {
    const { id, text, className } = button;

    if (!className) return;

    test(`${id} button to have ${className} className`, () => {
      const { getByTestId } = render(
        <Button id={id} handleClick={() => {}} className={className}>{text}</Button>,
      );

      expect(getByTestId(id)).toHaveClass(className);
    });
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<Button id="btn4" handleClick={() => {}}>4</Button>);

    expect(tree).toMatchSnapshot();
  });
});
