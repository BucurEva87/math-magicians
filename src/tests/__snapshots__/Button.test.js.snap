// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator interface buttons matches snapshot 1`] = `
<button
  className=""
  data-testid="btn4"
  id="btn4"
  onClick={[Function]}
  type="button"
>
  4
</button>
`;
